### PR TITLE
Add support for signal specific timeout and headers autoconfiguration.

### LINKED ISSUE #3430: Support OTLP signal specific timeout / headers configuration
**Is your feature request related to a problem? Please describe.**
N/A

**Describe the solution you'd like**
Currently, you can only specify `OTEL_EXPORTER_OTLP_TIMEOUT` and `OTEL_EXPORTER_OTLP_HEADERS`.

Per the [specification](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.5.0/specification/protocol/exporter.md):
- `OTEL_EXPORTER_OTLP_TRACES_TIMEOUT` and `OTEL_EXPORTER_OTLP_METRICS_TIMEOUT` support should be added.
- `OTEL_EXPORTER_OTLP_TRACES_HEADERS` and `OTEL_EXPORTER_OTLP_METRICS_HEADERS` support should be added.

**Describe alternatives you've considered**
N/A

**Additional context**
Support already exists for signal specific configuration of the OTLP endpoint via `OTEL_EXPORTER_OTLP_TRACES_ENDPOINT` and `OTEL_EXPORTER_OTLP_METRICS_ENDPOINT`.

I've also opened a separate ticket for adding [signal specific support for certificates](https://github.com/open-telemetry/opentelemetry-java/issues/3428). Adding support for the signal specific timeout and headers seems small enough for a single ticket.

### PR DESCRIPTION
**Pull request by jack-berg**
Resolves #3430.

### CODE REVIEW

File: sdk-extensions/autoconfigure/build.gradle.kts [hunk 1]
```diff
@@ -14,7 +14,7 @@ testSets {
     create("testInitializeRegistersGlobal")
     create("testJaeger")
     create("testPrometheus")
-    create("testOtlpTls")
+    create("testOtlp")
```
Comment (jack-berg) [inline sdk-extensions/autoconfigure/build.gradle.kts:17]: Previously there were OTLP specific tests in `testFullConfig` and `testOtlpTls`. I've renamed `testOtlpTls` => `testOtlp` and moved all OTLP specific tests there.

File: sdk-extensions/autoconfigure/build.gradle.kts [hunk 2]
```diff
@@ -97,11 +98,7 @@ tasks {
         environment("OTEL_BSP_SCHEDULE_DELAY", "10")
     }
 
-    val testOtlpTls by existing(Test::class) {
-        environment("OTEL_RESOURCE_ATTRIBUTES", "service.name=test,cat=meow")
-        environment("OTEL_TRACES_EXPORTER", "otlp")
-        environment("OTEL_METRICS_EXPORTER", "none")
-        environment("OTEL_BSP_SCHEDULE_DELAY", "10")
```
Comment (jack-berg) [inline sdk-extensions/autoconfigure/build.gradle.kts:104]: These env vars were unused in `testOtlpTls`.

File: sdk-extensions/autoconfigure/build.gradle.kts (hunk 1 above)
Comment (anuraaga) [reply sdk-extensions/autoconfigure/build.gradle.kts:17]: Thanks seems to have been some bad-tasting copy pasta

**Review by anuraaga — APPROVED** [review]

**Review by jkwatson — APPROVED** [review]
Thanks!

**Pull request merged**